pkg/block/blocktest/adapter.go

```diff
@@ -139,27 +139,33 @@ func testAdapterWalker(t *testing.T, adapter block.Adapter, storageNamespace str
 				return nil
 			})
 			require.NoError(t, err)
-			var prefix string
+
+			// prepare expectedResults
+			var (
+				expectedResults []string
+				prefix          string
+			)
 			if tt.prefix == "" {
 				if adapter.BlockstoreType() != block.BlockstoreTypeLocal {
 					prefix = testPrefix
 				}
-
-				require.Equal(t, path.Join(prefix, "folder_0.txt"), results[0])
-				results = results[1:]
+				expectedResults = append(expectedResults, path.Join(prefix, "folder_0.txt"))
 				for i := 0; i < filesAndFolders; i++ {
 					for j := 0; j < filesAndFolders; j++ {
-						require.Equal(t, path.Join(prefix, fmt.Sprintf("folder_%d/test_file_%d", i, j)), results[i*filesAndFolders+j])
+						expectedResults = append(expectedResults, path.Join(prefix, fmt.Sprintf("folder_%d/test_file_%d", i, j)))
 					}
 				}
 			} else {
 				if adapter.BlockstoreType() != block.BlockstoreTypeLocal {
 					prefix = tt.prefix
 				}
+
+				require.Lenf(t, results, filesAndFolders, "expected %d results, got %d", filesAndFolders, len(results))
```

Comment:
Ideally we would use deep.Equal here to compare filesAndFolders with results (or some function of results).  The current code gives bad error messages when it fails.

Reply:
I missed this line - built expected result and will diff between outside the 'if' block